libbeat/reader/readfile/timeout.go

```diff
@@ -68,9 +70,13 @@ func (r *TimeoutReader) Next() (reader.Message, error) {
 		go func() {
 			for {
 				message, err := r.reader.Next()
-				r.ch <- lineMessage{message, err}
-				if err != nil {
-					break
+				select {
+				case <-r.done:
+					return
+				case r.ch <- lineMessage{message, err}:
+					if err != nil {
+						break
```

Comment:
this break will not exit the loop anymore. It is now scoped by 'select'. Better use `return`